### ISSUE TITLE
Implement noex::Table::AsyncSampleRows

### ISSUE THREAD

**Issue by coryan** [GoogleCloudPlatform/google-cloud-cpp#1238]
The API should (roughly) look like this:

```C++
template <typename Functor, ...>
void AsyncSampleRows(CompletionQueue& cq, Functor&& callback);
```

The callback should receive a `std::vector<bigtable::RowKeySample>` parameter.

**Comment by coryan** [comment]
A word of caution: this function should return a consistent view of the sample, if the sampling fails and the function retries it should clear the samples collected so far and start over.

**Comment by dopiera** [comment]
Fixed by #1369
